### PR TITLE
[Fix/Train/Model] debug the fp16 scale issue for loss backward

### PR DESCRIPTION
**Pull request by ZhangYuanhan-AI**
1. debug the fp16 scale issue for loss backward 
2. debug the label mask problem (mask for the whole batch, instead of only the first item in the batch)

### CODE REVIEW

File: pipeline/train/instruction_following.py [hunk 1]
```diff
@@ -123,7 +125,11 @@ def train_one_epoch(args, model, epoch, mimicit_loaders, tokenizer, optimizer, l
                 #     attention_mask=attention_mask.to(device_id),
                 #     max_length=256,
                 # )
-            accelerator.backward(loss_mimicit)
+            if accelerator.mixed_precision != "bf16":
```
Comment (Luodian) [inline pipeline/train/instruction_following.py]: From my understanding. This should be judged by if we use `mix_precision`, not actually by if `mix_precision=bf16`. Since the loss is calculated within a `autocast()`.

**Review by Luodian — APPROVED** [review]
